packages/database/test/exp/integration.test.ts

```diff
@@ -257,26 +260,28 @@ describe('Database@exp Tests', () => {
     expect(events[0]).to.equal('a');
   });
 
-  it('Can goOffline/goOnline', async () => {
-    const db = getDatabase(defaultApp);
-    goOffline(db);
-    try {
-      await get(ref(db, 'foo/bar'));
-      expect.fail('Should have failed since we are offline');
-    } catch (e) {
-      expect(e.message).to.equal('Error: Client is offline.');
-    }
-    goOnline(db);
-    await get(ref(db, 'foo/bar'));
-  });
-
   it('Can delete app', async () => {
     const db = getDatabase(defaultApp);
     await deleteApp(defaultApp);
     expect(() => ref(db)).to.throw('Cannot call ref on a deleted database.');
     defaultApp = undefined;
   });
 
+  it('waits until the database is online to resolve the get request', async () => {
+    const db = getDatabase(defaultApp);
+    const r = ref(db, 'foo2');
+    const initial = {
+      test: 1
+    };
+    await set(r, initial);
+    goOffline(db);
+    const getValue = promiseWithTimeout(get(r));
+    expect(getValue).to.be.rejectedWith('timeout!');
+    goOnline(db);
+    const deferredTimeout2 = await promiseWithTimeout(get(r));
+    expect(deferredTimeout2.val()).to.deep.equal(initial);
+  });
+
```

Comment:
Can we also add a test that does this:
- Go offline
- Place a get() call
- Go online
- Expect get to complete